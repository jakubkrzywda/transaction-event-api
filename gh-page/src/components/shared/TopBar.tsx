import React from "react";
import TopBarLogo from "./TopBarLogo";

const TopBar = () => {
    return (
        <div className="app-header">
            <TopBarLogo />
        </div>
    );
};

export default TopBar;