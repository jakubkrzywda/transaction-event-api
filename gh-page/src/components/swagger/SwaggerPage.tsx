import React from "react";
import SwaggerUI from "swagger-ui-react";
import "swagger-ui-react/swagger-ui.css";
import TopBar from "../shared/TopBar";
import { ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

let yam = require("./api.yaml");

process.env["NODE_TLS_REJECT_UNAUTHORIZED"] = "0"

const SwaggerPage = () => {
    return (
        <>
            <TopBar />
            <SwaggerUI url={window.location.origin + yam}
                docExpansion="list"
                defaultModelExpandDepth={1} />
            <ToastContainer />
        </>);
};

export default SwaggerPage;
