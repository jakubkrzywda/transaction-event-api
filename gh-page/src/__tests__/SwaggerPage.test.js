import React from "react";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import SwaggerPage from "../components/swagger/SwaggerPage";
import TopBar from "../components/shared/TopBar";
import yam from "../components/swagger/api.yaml";
import Base64Reader from "../components/swagger/Base64Reader";

Enzyme.configure({adapter: new Adapter()});

describe("SwaggerPage", () => {
    it("renders without crashing", () => {
        const wrapper = shallow(<SwaggerPage />);
        expect(wrapper.exists()).toBe(true);
    });
  
    it("matches snapshot", () => {
        const wrapper = shallow(<SwaggerPage />);
        expect(wrapper).toMatchSnapshot();
    });
    
    it("renders topbar", () => {
        const wrapper = shallow(<SwaggerPage />);
        expect(wrapper.contains(<TopBar/>));
    });
  
    it("renders swagger", () => {
        const wrapper = shallow(<SwaggerPage />);
        expect(wrapper.find("SwaggerUI")).toBeDefined();
    });
    
    it("renders swagger with correct yaml", () => {
        const wrapper = shallow(<SwaggerPage />);
        expect(wrapper.find("SwaggerUI").prop("url")).toBe(window.location.origin + yam)
    });
    
    it("renders swagger with docexpansion equalto list", () => {
        const wrapper = shallow(<SwaggerPage />);
        expect(wrapper.find("SwaggerUI").prop("docExpansion")).toBe("list");
    });
    
    it("renders swagger with defaultModelExpandDepth equalto 1", () => {
        const wrapper = shallow(<SwaggerPage />);
        expect(wrapper.find("SwaggerUI").prop("defaultModelExpandDepth")).toBe(1);
    });
});