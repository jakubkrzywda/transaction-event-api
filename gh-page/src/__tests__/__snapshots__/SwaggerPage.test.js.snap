// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwaggerPage matches snapshot 1`] = `ShallowWrapper {}`;
